File: Estetica Regenere/Providers/LoginProvider/LoginProvider.m
//
//  LoginProvider.m
//  Estetica Regenere
//
//  Created by Adriano on 10/14/15.
//  Copyright (c) 2015 Adriano. All rights reserved.
//

#import "LoginProvider.h"
#import "AFHTTPRequestOperationManager.h"

@implementation LoginProvider

-(void)performLoginWithEmail:(NSString *)email
                     passord:(NSString *)password
                    delegate:(id<LoginProviderDelegate>)delegate
{
    NSDictionary *body = @{
                           @"email": email,
                           @"password": password
                           };
    
    [self performRequestWithPath:@"/login/"
                          headers:@{}
                             data:body
                           method:@"POST"
                        callback:delegate];
}


-(void)handleSuccesfulResponse:(NSDictionary *)response callback:(id)callback
{
    [callback onLoginSuccessWithToken:response[@"content"][@"token"]];
}

-(void)handleFailedResponse:(NSDictionary *)response callback:(id)callback
{
    [callback onLoginFailure];
}


@end

File: app/src/main/java/com/example/adriano/esteticaregenere_android/Providers/LoginProvider.java
package com.example.adriano.esteticaregenere_android.Providers;

import android.content.Context;

import com.loopj.android.http.RequestParams;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Adriano on 12/31/15.
 */
public class LoginProvider extends BaseProvider {

    public LoginProvider() {}

    public void performLogin(Context context, String login, String password, LoginProviderCallback callback) {
        JSONObject body = new JSONObject();
        try {
            body.put("email", login);
            body.put("password", password);
            post(context, "login/", body, callback);
        } catch (JSONException e) {
            e.printStackTrace();
            callback.onResponseFailure();
        }

    }

    @Override
    void handleSuccessfulResponse(JSONObject response) {
        try {
            ((LoginProviderCallback)callback).onLoginSuccess(response.getJSONObject("content").getString("token"));
        } catch (JSONException e) {
            e.printStackTrace();
            callback.onResponseFailure();
        }
    }

    @Override
    void handleFailedResponse(JSONObject response, int statusCode) {
        ((LoginProviderCallback)callback).onLoginFailure();
    }
}
